### PR TITLE
[dice] Refactor some dice related constants and APIs

### PR DESCRIPTION
**Pull request by tommychiu-github**
Some DICE key creation constants and utilities are invoked by several modules.
Refactor them for a better code reuse.

### CODE REVIEW

**Comment by tommychiu-github** [comment]
@timothytrippel for viz.

File: sw/device/silicon_creator/lib/cert/cbor.c [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+    }                                   \
+  } while (0)
+
+#define CBOR_CHECK_OVERFLOWED_AND_RETURN(p) \
+  do {                                      \
+    CBOR_RETURN_IF_OVERFLOWED(p);           \
+    return kErrorOk;                        \
+  } while (0)
+
+inline rom_error_t cbor_write_out_init(struct CborOut *p, void *buf,
```
Comment (timothytrippel) [inline sw/device/silicon_creator/lib/cert/cbor.c:28]: I think you want to keep all of these inline functions in the header file and mark then `static inline`. You can still share them with everything that include the header. It ensure they get inlined without LTO: https://stackoverflow.com/a/47821267 (applies to all inline functions)

File: sw/device/silicon_creator/lib/cert/cbor.h [hunk 2]
```diff
@@ -6,100 +6,52 @@
...
-  CBOR_CHECK_OVERFLOWED_AND_RETURN(p);
-}
-
-static inline rom_error_t cbor_write_bytes(struct CborOut *p,
-                                           const uint8_t *data,
-                                           const size_t data_size) {
-  CborWriteBstr(data_size, data, p);
-  CBOR_CHECK_OVERFLOWED_AND_RETURN(p);
-}
+rom_error_t cbor_write_out_init(struct CborOut *p, void *buf,
```
Comment (timothytrippel) [inline sw/device/silicon_creator/lib/cert/cbor.h:12]: all functions prototypes in the header file should have doxygen style comments: https://cs.opensource.google/opentitan/opentitan/+/master:sw/device/lib/testing/i2c_testutils.h;l=158

Comment (timothytrippel) [inline sw/device/silicon_creator/lib/cert/cbor.h:12]:
```suggestion
OT_WARN_UNUSED
rom_error_t cbor_write_out_init(struct CborOut *p, void *buf,
```
?

File: sw/device/silicon_creator/lib/cert/BUILD [hunk 3]
```diff
@@ -130,6 +131,17 @@ cc_library(
         "//hw/ip/otp_ctrl/data:otp_ctrl_c_regs",
         "//sw/device/lib/base:status",
         "//sw/device/silicon_creator/lib:attestation",
+        "//sw/device/silicon_creator/lib/cert:dice_keys",
+        "//sw/device/silicon_creator/manuf/lib:flash_info_fields",
```
Comment (timothytrippel) [inline sw/device/silicon_creator/lib/cert/BUILD:135]: why are these two deps needed here?

File: sw/device/silicon_creator/lib/base/util.h [hunk 4]
```diff
@@ -12,6 +12,23 @@
...
+   * Size of an attestation signature component in bits.
+   */
+  kUtilEcdsaP256SignatureComponentBits = 256,
+  /**
+   * Size of an attestation signature component in bytes.
+   */
+  kUtilEcdsaP256SignatureComponentBytes =
+      kUtilEcdsaP256SignatureComponentBits / 8,
+  /**
+   * Size of an attestation signature component in 32b words.
```
Comment (timothytrippel) [inline sw/device/silicon_creator/lib/base/util.h:26]:
```suggestion
   * Size of an ECDSA signature component in 32b words.
```

File: sw/device/silicon_creator/lib/base/util.h [hunk 5]
```diff
@@ -12,6 +12,23 @@
 extern "C" {
 #endif
 
+enum {
+  /**
+   * Size of an attestation signature component in bits.
+   */
+  kUtilEcdsaP256SignatureComponentBits = 256,
+  /**
+   * Size of an attestation signature component in bytes.
```
Comment (timothytrippel) [inline sw/device/silicon_creator/lib/base/util.h:21]:
```suggestion
   * Size of an ECDSA signature component in bytes.
```

File: sw/device/silicon_creator/lib/base/util.h [hunk 6]
```diff
@@ -12,6 +12,23 @@
 extern "C" {
 #endif
 
+enum {
+  /**
+   * Size of an attestation signature component in bits.
```
Comment (timothytrippel) [inline sw/device/silicon_creator/lib/base/util.h:17]:
```suggestion
   * Size of an ECDSA signature component in bits.
```

File: sw/device/silicon_creator/lib/cert/cbor.h (hunk 2 above)
Comment (tommychiu-github) [reply sw/device/silicon_creator/lib/cert/cbor.h:12]: Yep, I think we'll need this enforcement in all the wrappers here.

**Review by timothytrippel — APPROVED** [review]

**Pull request merged**